### ISSUE TITLE
Calling add_control_to_bottom_panel can crash engine

### ISSUE THREAD

**Issue by GlaDOSik** [godotengine/godot#19605]
**Godot version:**
Godot 3.0.3 stable

**OS/device including version:**
Windows 7

**Issue description:**
Setting anything else than Control using method EditorPlugin.add_control_to_bottom_panel will crash engine. It's not a big deal but it shouldn't crash the engine. Probably a good bug for label junior job.

**Steps to reproduce:**
Create a plugin using basic template skeleton and call add_control_to_bottom_panel with null or anything else than Control. The engine will crash.

**Comment by Ranoller** [comment]
This should give a compiler error like other plugin mistakes and return line... So this function should check if argument inherits control, i think.

**Comment by leecommamichael** [comment]
I attempted to reproduce and I receive an error without crash:
``SCRIPT ERROR: _enter_tree: Invalid type in function 'add_control_to_dock' in base 'EditorPlugin (custom_dock.gd)'. Cannot convert argument 2 from Object to Object.
   At: res://addons/my_custom_dock/custom_dock.gd:12.``
In this case argument 2 is the preloaded dock.tscn with an ordinary Node at the root and no other nodes. Upon making the plugin inactive through **Project Settings**, I receive another error without crash: ``SCRIPT ERROR: _exit_tree: Invalid type in function 'remove_control_from_docks' in base 'EditorPlugin (custom_dock.gd)'. Cannot convert argument 1 from Object to Object.
   At: res://addons/my_custom_dock/custom_dock.gd:18.``

The errors aren't helpful, but they don't seem to crash. Tested on a fresh build of ``master`` just before writing, Fedora 28, x64.

**Comment by leecommamichael** [comment]
I'd like to resolve the issue. I don't think we can keep the user from making the mistake since GDScript seems not to have dynamic type checking. Without significant change I added a warning message to the editor output, which seems to fix the crash. Is that sufficient?

**Comment by akien-mga** [comment]
@leecommamichael Sounds good, can you make a pull request so that we can review it?

**Comment by akien-mga** [comment]
See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html if needed.

**Comment by leecommamichael** [comment]
@akien-mga Thanks for the resources. It appears that ``EditorPlugin::add_control_to_bottom_panel(...)`` is not called in the event that there is a SCRIPT_ERROR generated. I failed to reproduce the crash without explicitly passing null, I tested with scenes containing a root of Node and Sprite without crash. I could simply write ``ERR_FAIL_NULL_V(p_control, nullptr);`` to prevent a crash on null. My instinct is that checking the type hierarchy is probably not EditorPlugin's job, and more of a script compilation problem. However I know not to assume how the primary maintainers feel.
